PSICompile/ILCodeGen.cs

```diff
@@ -122,7 +132,10 @@ public class ILCodeGen : Visitor {
       u.Expr.Accept (this);
       string op = u.Op.Kind.ToString ().ToLower ();
       op = op switch { "sub" => "neg", _ => op };
-      Out ($"    {op}");
+      if (op is "not") {
+         Out ("    ldc.i4.0");
+         Out ("    ceq");
+      } else Out ($"    {op}");

```

Comment:
This works correctly only for boolean values. So this works:
`WriteLn (not false)` should print **True.**
But this does not work:
`WriteLn (not 2)` should print **-3**
Because the not of an integer value is the integer you get by flipping all the bits in the binary representation. So the code you generate for not should depend on the data type of the expression.

Reply:
implemented "not" operator by checking the data type of expression:

![image](https://github.com/tarydon/PSI/assets/125174306/283fcbb2-342c-4329-b928-2a52670b24ca)